File: Data Structures & Algorithms/three-integer-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        vector<vector<int>> res = {};

        std::sort(nums.begin(), nums.end());

        for(int i{0}; i < nums.size(); i++){
            if(i > 0 && nums[i] == nums[i-1]){
                continue;
            }

            int l = i + 1, r = nums.size() - 1;
            while(l < r){
                int threesum = nums[i] + nums[l] + nums[r];
                if(threesum > 0){
                    r -= 1;
                }else if(threesum < 0){
                    l+=1;
                }else{
                    res.push_back({nums[i], nums[l], nums[r]});
                    l += 1;
                    while (nums[l] == nums[l-1] && l < r) {
                        l += 1;
                    }
                }
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        bucket = [ [] for _ in range(len(nums) + 1)]
        freq = {}

        for n in nums:
            freq[n] = 1 + freq.get(n, 0)

        for num, f in freq.items():
            bucket[f].append(num)

        res = []
        for i in range(len(bucket) - 1, 0, -1):
            for num in bucket[i]:
                res.append(num)
                if len(res) == k:
                    return res